common/src/main/scala/org/apache/celeborn/common/meta/WorkerInfo.scala

```diff
@@ -261,30 +271,46 @@ object WorkerInfo {
     new WorkerInfo(host, rpcPort.toInt, pushPort.toInt, fetchPort.toInt, replicatePort.toInt)
   }
 
-  def fromPbWorkerInfo(pbWorker: PbWorkerInfo): WorkerInfo = {
+  def fromPbWorkerInfo(pbWorkerInfo: PbWorkerInfo): WorkerInfo = {
     val disks =
-      if (pbWorker.getDisksCount > 0) {
-        pbWorker.getDisksList.asScala
+      if (pbWorkerInfo.getDisksCount > 0) {
+        pbWorkerInfo.getDisksList.asScala
           .map { pbDiskInfo => pbDiskInfo.getMountPoint -> PbSerDeUtils.fromPbDiskInfo(pbDiskInfo) }
           .toMap.asJava
       } else {
         new util.HashMap[String, DiskInfo]()
       }
+    val userResourceConsumption = pbWorkerInfo
+      .getUserResourceConsumptionMap
+      .asScala
+      .map { case (userInfo, pbResourceConsumption) =>
+        (UserIdentifier(userInfo), pbResourceConsumption)
+      }
+      .mapValues(PbSerDeUtils.fromPbResourceConsumption)
+      .asJava
```

Comment:
Can we extract these codes? I saw it was used in many places

Reply:
> Can we extract these codes? I saw it was used in many places

Agree, and there are exist methods in MetaUtil, but they use the Resources.proto rather than TransportMessages.proto.
```
public static Map<UserIdentifier, ResourceConsumption> fromPbUserResourceConsumption(
      Map<String, ResourceProtos.ResourceConsumption> pbUserResourceConsumption) {
    Map<UserIdentifier, ResourceConsumption> map = new HashMap<>();
    pbUserResourceConsumption.forEach(
        (k, v) -> {
          ResourceConsumption resourceConsumption =
              new ResourceConsumption(
                  v.getDiskBytesWritten(),
                  v.getDiskFileCount(),
                  v.getHdfsBytesWritten(),
                  v.getHdfsFileCount());
          map.put(ControlMessages.UserIdentifier$.MODULE$.apply(k), resourceConsumption);
        });
    return map;
  }
```
 I will extract similar methods in PbSerDeUtils for TransportMessages's class, and we can replace MetaUtil's method  with them directly in future when drop the Resources.proto.